tools/src/test/scala/com/nvidia/spark/rapids/tool/planparser/SqlPlanParserSuite.scala

```diff
@@ -52,50 +101,197 @@ class SQLPlanParserSuite extends FunSuite with BeforeAndAfterEach with Logging {
+    }
+    val allExecInfo = getAllExecsFromPlan(parsedPlans.toSeq)
+    val parquet = {
+      allExecInfo.filter(_.exec.contains("CreateDataSourceTableAsSelectCommand"))
+    }
+    assertSizeAndNotSupported(1, parquet.toSeq)
+  }
+
+  test("InMemoryTableScan") {
+    TrampolineUtil.withTempDir { eventLogDir =>
+      val (eventLog, _) = ToolTestUtils.generateEventLog(eventLogDir,
+        "InMemoryTableScan") { spark =>
+        import spark.implicits._
+        val df = spark.sparkContext.makeRDD(1 to 10000, 6).toDF
+        val dfc = df.cache()
+        dfc
+      }
+      val pluginTypeChecker = new PluginTypeChecker()
+      val app = createAppFromEventlog(eventLog)
+      assert(app.sqlPlans.size == 1)
+      val parsedPlans = app.sqlPlans.map { case (sqlID, plan) =>
+        SQLPlanParser.parseSQLPlan(plan, sqlID, pluginTypeChecker, app)
       }
+      val allExecInfo = getAllExecsFromPlan(parsedPlans.toSeq)
+      val tableScan = allExecInfo.filter(_.exec == ("InMemoryTableScan"))
+      assertSizeAndSupported(1, tableScan.toSeq)
+    }
+  }
+
+  test("BroadcastExchangeExec, SubqueryBroadcastExec and Exchange") {
+    val profileLogDir = ToolTestUtils.getTestResourcePath("spark-events-qualification")
+    val eventLog = s"$profileLogDir/nds_q86_test"
+    val pluginTypeChecker = new PluginTypeChecker()
+    val app = createAppFromEventlog(eventLog)
+    assert(app.sqlPlans.size > 0)
+    val parsedPlans = app.sqlPlans.map { case (sqlID, plan) =>
+      SQLPlanParser.parseSQLPlan(plan, sqlID, pluginTypeChecker, app)
+    }
+    val allExecInfo = getAllExecsFromPlan(parsedPlans.toSeq)
+    val broadcasts = allExecInfo.filter(_.exec == "BroadcastExchange")
+    assertSizeAndSupported(3, broadcasts.toSeq, Seq(Some(1154), Some(1154), Some(1855)))
+    val subqueryBroadcast = allExecInfo.filter(_.exec == "SubqueryBroadcast")
+    assertSizeAndSupported(1, subqueryBroadcast.toSeq, Seq(Some(1175)))
+    val exchanges = allExecInfo.filter(_.exec == "Exchange")
+    assertSizeAndSupported(2, exchanges.toSeq, Seq(Some(15688), Some(8)))
+  }
+
+  test("CustomShuffleReaderExec") {
+    val profileLogDir = ToolTestUtils.getTestResourcePath("spark-events-qualification")
+    // this is eventlog because CustomShuffleReaderExec only available before 3.2.0
+    val eventLog = s"$profileLogDir/customshuffle_eventlog.zstd"
+    val pluginTypeChecker = new PluginTypeChecker()
+    val app = createAppFromEventlog(eventLog)
+    assert(app.sqlPlans.size > 0)
+    val parsedPlans = app.sqlPlans.map { case (sqlID, plan) =>
+      SQLPlanParser.parseSQLPlan(plan, sqlID, pluginTypeChecker, app)
+    }
+    val allExecInfo = getAllExecsFromPlan(parsedPlans.toSeq)
+    val reader = allExecInfo.filter(_.exec == "CustomShuffleReader")
+    assertSizeAndSupported(2, reader.toSeq)
+  }
+
+  test("AQEShuffleReadExec") {
+    val profileLogDir = ToolTestUtils.getTestResourcePath("spark-events-qualification")
```

Comment:
We could use `logDir` which is already a class variable. Also, maybe renaming it to `qualLogdir` would be better.

Reply:
updated